src/Storages/StorageS3Cluster.cpp

```diff
@@ -10,6 +10,7 @@
 #include <DataTypes/DataTypeString.h>
 #include <IO/WriteBufferFromS3.h>
 #include <IO/WriteHelpers.h>
+#include <IO/ConnectionTimeoutsContext.h>
```

Comment:
#include seems unrelated

Reply:
Compile error on s390x. Seems missing header file for function getTCPTimeoutsWithFailover(), not sure why this only appears on s390x. 
```./src/Storages/StorageS3Cluster.cpp:130: undefined reference to `DB::ConnectionTimeouts::getTCPTimeoutsWithFailover(DB::Settings const&)' ```

Reply:
Somehow the style check complains that this #include is duplicated: https://s3.amazonaws.com/clickhouse-test-reports/45520/dea7e2e1f4d5700efb1a339ede45b17c837c2c1b/style_check/duplicate_includes_output.txt

(because of that, the "merge" button is also greyed out for me)

I think we need to get rid of the include. What I don't understand is if you get a compiler error or a link error on s390x. The error you pasted is a link error for sure.

What I see is that function `getTCPTimeoutsWithFailover()` is declared in IO/ConnectionTimeouts.h and defined in IO/ConnectionTimeoutsContext.h, as an `inline` function in a header file though. This means that depending on whether
- the linker actually inlined the code, it will work (and that seems to be the case in ClickHouse CI) or 
- the linker did not inline the code, it will work only if StorageS3Cluster.cpp's library links the library which contains IO/ConnectionTimeoutsContext.h.

This is a general issue which I will try to fix separately.

As a workaround, you could try if replacing `inline` by `ALWAYS_INLINE` in IO/ConnectionTimeoutsContext.h will help.

Reply:
After using `ALWAY_INLINE` instead of `inline`, got multiple definition error. I believe this is a linker error. This will probably take a while to debug.

> multiple definition of DB::ConnectionTimeouts::getTCPTimeoutsWithFailover(DB::Settings const&); src/Functions/CMakeFiles/clickhouse_functions_obj.dir/./catboostEvaluate.cpp.o:catboostEvaluate.cpp:(.text+0x0): first defined here


Can we do something like the following just to make sure it builds for now?
```suggestion
#if defined(__s390x__)
#include <IO/ConnectionTimeoutsContext.h>
#endif
```

Reply:
That's okay.

Alternatively, wait for #46006 which will fix the underlying problem.

Reply:
Since you merged master into this PR and, as a result, got #46006, you would need to remove `#include <IO/ConnectionTimeoutsContext.h>` from this file. ConnectionTimeoutsContext was removed in that other PR.

(I obviously don't have permissions to push to repo clibmouse, otherwise I would have done this little change by myself)

Reply:
```suggestion
```